applications/nrf_desktop/src/modules/ble_scan.c

```diff
@@ -654,6 +664,28 @@ static bool app_event_handler(const struct app_event_header *aeh)
 		return false;
 	}
 
+	if (IS_ENABLED(CONFIG_DESKTOP_BLE_SCAN_PM_EVENTS) &&
+	    is_power_down_event(aeh)) {
+		if (scanning) {
+			scan_stop();
+		}
+
+		scan_blocked = true;
+		k_work_cancel_delayable(&scan_start_trigger);
+
+		return false;
+	}
+
+	if (IS_ENABLED(CONFIG_DESKTOP_BLE_SCAN_PM_EVENTS) &&
+	    is_wake_up_event(aeh)) {
+		if (scan_blocked) {
+			scan_blocked = false;
```

Comment:
Now we could start scanning before settings are loaded. We could get power_down + wake_up sequence before ble_bond is ready. That could happen when USB suspend/resume happens right after boot.

I think we will need to introduce a proper state machine to ensure that all of the cases are handled properly.